contrib/pkg/adm/managedns/enable.go

```diff
@@ -191,15 +206,59 @@ func (o *Options) Run(dynClient client.Client, args []string) error {
 	return nil
 }
 
-func podHasEnvVar(p corev1.Pod, envVar string) bool {
-	for _, ev := range p.Spec.Containers[0].Env {
-		if ev.Name == envVar {
-			return true
+func getUpdatedManagedDNSConfig(dynClient client.Client, previousGeneration int64) (string, error) {
+	deployment := &appsv1.Deployment{}
+	key := types.NamespacedName{Name: hiveAdmissionDeployment, Namespace: hiveNamespace}
+
+	log.Info("waiting for new hiveadmission deployment")
+	for i := 0; i < 24; i++ {
+		err := dynClient.Get(context.TODO(), key, deployment)
+		if err != nil {
+			log.WithError(err).Fatal("error getting hiveadmission deployment")
+			return "", err
+		}
+
+		if deployment.Generation > previousGeneration {
+			for i, vol := range deployment.Spec.Template.Spec.Volumes {
+				if vol.Name == constants.ManagedDomainsVolumeName {
+					return deployment.Spec.Template.Spec.Volumes[i].ConfigMap.Name, nil
+				}
+			}
+		}
+
+		log.Info("new hiveadmission deployment not ready, sleeping 5s")
+		time.Sleep(5 * time.Second)
+	}
+
+	log.Error("waited for new hiveadmission deployment, but it never appeared")
+	return "", fmt.Errorf("waited for new hiveadmission deployment, but it never appeared")
+}
+
+func podUsingConfigMap(p corev1.Pod, cmName string) bool {
+	for _, vol := range p.Spec.Volumes {
+		if vol.Name == constants.ManagedDomainsVolumeName {
+			if vol.ConfigMap.Name == cmName {
```

Comment:
Can this be `return vol.ConfigMap.Name == cmName`? Is there an expectation that there could be multiple volumes with the name of `ManagedDomainsVolumeName`?

Reply:
I would hope not. I'll fix it up to just do the equality return immediately.